auth/kerberos/src/krb.cpp

```diff
@@ -737,14 +759,23 @@ bool is_ticket_ready_for_renewal( std::string krb_cc_name )
     boost::split( results, krb_ticket_info_result.second, []( char c ) { return c == '#'; } );
     std::string renew_until = "renew until";
     bool is_ready_for_renewal = false;
-
+    
     for ( auto& result : results )
     {
         auto found = result.find( renew_until );
         if ( found != std::string::npos )
         {
             found += renew_until.length();
-            std::string renewal_date_time = result.substr( found + 1, result.length() );
+            std::string renewal_date_time;
+            
+            if (krb_ticket_info->origin == "file")
```

Comment:
what does origin means, this is a breaking change. Can we use any other existing fields?

Reply:
@ford-th 
> Currently the AWS master branch uses "token renewal expiration" which seems incorrect since that expires long after the actual token expires (We found this issue when running in our UAT environment).

Can you share any logs or script. If you have a script, that can be added as a regression script in this repo.

Reply:
Hi Tom, having the "origin" property as a nullable field might help. We scheduled a quick meeting to discuss to see if we can leverage any existing fields. If we need this field, lets also discuss how to add this without breaking for existing customers.

Reply:
@smhmhmd @saikiranakula-amzn The origin property has been removed from the PR and the code know looks at the token expiration instead of the refresh token expiration